services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -1357,6 +1366,41 @@ private void deleteStore(
     }
   }
 
+  /**
+   * Deletes the acls associated with a store
+   */
+  protected void cleanupAclsForStore(Store store, String storeName, String clusterName) {
+    if (store == null) {
+      LOGGER.warn("Store object for: {} is missing in cluster: {}! Skipping acl deletion!", storeName, clusterName);
+      return;
+    }
+    if (store.isMigrating()) {
+      LOGGER.info("Store: {} in cluster: {} is migrating! Skipping acl deletion!", storeName, clusterName);
+      return;
+    }
+    if (!getAuthorizerService().isPresent()) {
+      LOGGER.info(
+          "Authorizer service is not present! Skipping acl deletion for store: {} in cluster: {}",
+          storeName,
+          clusterName);
+      return;
+    }
```

Comment:
```suggestion
    if (store == null) {
      LOGGER.warn("Store object for: {} is missing in cluster: {}! Skipping acl deletion!", storeName, clusterName);
      return;
    } else if (store.isMigrating()) {
      LOGGER.info("Store: {} in cluster: {} is migrating! Skipping acl deletion!", storeName, clusterName);
      return;
    } else if (!getAuthorizerService().isPresent()) {
      LOGGER.info(
          "Authorizer service is not present! Skipping acl deletion for store: {} in cluster: {}",
          storeName,
          clusterName);
      return;
    }
```
super nit: we can save some lines here

Reply:
Given that each block exits the method, the else if chain isn’t needed.
Flattening the structure improves readability and keeps the flow simpler IMHO